components/tikv_util/src/future_pool/builder.rs

```diff
@@ -50,15 +53,17 @@ impl Builder {
     where
         F: Fn() + Send + Sync + 'static,
     {
-        self.inner_builder.before_stop(f);
+        self.mark |= 1;
+        self.inner_builder.before_stop_wrapper(f);
```

Comment:
I think you can store `f` as a member of `Builder` and then pass it to `inner_builder` in `build()`.

Reply:
emm..I think a mark variable is much smaller than two `f` in `after_start` and `before_stop`.🤣

Reply:
I think `mark` is opaque, could you give a meaningful name or leave some comments?

Reply:
Why should it be _smaller_? I think the overhead is trivial. Readability should have a high priority.